ABC related modifications (#255)
XanaduAI/MrMustard: .github/CHANGELOG.md

* When projecting a Gaussian state onto a Fock state, the upper limit of the autocutoff now respect the Fock projection.
  [(#246)](https://github.com/XanaduAI/MrMustard/pull/246)

> * Change the internal function to calculate ABC of a Bargmann representation into the correct ABC corresponds to the litterature and some other fix to get the correct Fock tensor. [(#255)](https://github.com/XanaduAI/MrMustard/pull/255)

Reviewer: ```suggestion
* Fixed the internal function to calculate ABC of Bargmann representation (now corresponds to the literature) and other fixes to get the correct Fock tensor. [(#255)](https://github.com/XanaduAI/MrMustard/pull/255)
```
